### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by nbumbarger**

### CODE REVIEW

File: app/assets/scripts/components/map.js [hunk 1]
```diff
@@ -261,7 +261,13 @@ const Map = React.createClass({
 
   renderOverlayLegend: function (scale) {
     const isQuantile = scale.hasOwnProperty('invertExtent');
-    const iterable = (isQuantile ? scale.range() : scale.domain()).sort();
+    let iterable = (isQuantile ? scale.range() : scale.domain()).sort();
+    const category = this.props.overlay.category.toLowerCase();
```
Comment (dereklieu) [inline app/assets/scripts/components/map.js]: If for some reason an indicator didn't have an assigned category, this would throw an error as it would try to lower case an undefined value.

A better way would be:

```(javascript)
let category = get(this.props, 'overlay.category');
if (category) {
  category = category.toLowerCase()
  // do the rest of function
}
```

File: app/assets/scripts/components/map.js [hunk 2]
```diff
@@ -261,7 +261,13 @@ const Map = React.createClass({
...
+    const lookup = {'very low': 5, 'low': 4, 'medium': 3, 'high': 2, 'very high': 1, 5: 'very low', 4: 'low', 3: 'medium', 2: 'high', 1: 'very high'};
+    if (category === 'categorical') {
+      iterable = iterable.map((cat) => lookup[cat]).sort();
```
Comment (dereklieu) [inline app/assets/scripts/components/map.js]: If I'm understanding, this would just translate 'very low' to 5, and vice versa? 

So if the categories from the API were `[1, 4, 5]` you would get `['very high', 'low', 'very low']`, which would not sort in the way you want it to.

File: app/assets/scripts/components/page-footer.js [hunk 3]
```diff
@@ -28,13 +28,13 @@ var PageFooter = React.createClass({
                 <p className='updates__description'>Get notifications when we update projects <span className='italic'>quarterly</span>.</p>
                 <div id="mc_embed_signup">
                   <form action="//github.us14.list-manage.com/subscribe/post?u=7a02eb473bd111aa85bfb7ffc&amp;id=120d44b3d2" method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" className="validate form__input-group" target="_blank" noValidate>
-                        <input type="email" value={this.state.email} placeholder="Enter email address" name="EMAIL" className="required email form__control form__control--large" id="form-input-6" onChange={this.onChange}/>
+                      <input type="email" value={this.state.email} placeholder="Enter email address" name="EMAIL" className="required email form__control form__control--large" id="form-input-6" onChange={(e) => this.onChange(e)}/>
```
Comment (dereklieu) [inline app/assets/scripts/components/page-footer.js]: this should just be `onChange={this.onChange}`

File: app/assets/scripts/views/project-browse.js [hunk 4]
```diff
@@ -477,13 +488,17 @@ var ProjectBrowse = React.createClass({
...
-      activeProjectFilters.forEach((filter) => {
-        projects = projects.filter(filter.filter);
-      });
+    let projects = [];
+    let markers = [];
+    if (!this.state.projectsHidden) {
+      projects = this.props.api.projects;
+      if (activeProjectFilters.length) {
+        activeProjectFilters.forEach((filter) => {
+          projects = this.state.projectsHidden ? [] : projects.filter(filter.filter);
```
Comment (dereklieu) [inline app/assets/scripts/views/project-browse.js]: I'm not understanding why you need to check the `projectsHidden` boolean here if there's already a check for it above?

**Comment by dereklieu** [comment]
@nbumbarger mostly looks good, my main question is on how the category mapping for `low, very low, etc` is meant to work. The main issue from where I'm standing is, the categories are in theory allowed to be any value from the API. If they are values outside of the mappings you've hard-coded, they would all show up as `undefined`. 

Even if we expect that we're getting 1-5 now, because of the admin design we can't just naively hardcode this in. One possible check would be similar to the `isNumerical` check, where I check if all the indicator values can be successfully coerced to a number before calling it a sequential or divergent category. A similar check here would be to see if *all* the category values have a mapping, ie `1 to 'very low'`, before going forward with that mapping. 

Also it's possible that the category numbers come from the API as strings, ie `'1'` rather than `1`. I have not confirmed this though.

TLDR, it doesn't jive well if the admin allows any category but the front-end allows only a very narrow subset. cc @felskia

**Comment by nbumbarger** [comment]
@dereklieu I labeled that commit as a hack because I wasn't 100% comfortable with it either, but I checked with Ali and she thought that all categorical submissions would use low, medium, high etc. If we encounter a value outside of the mapping, should it just use the native values?

**Comment by dereklieu** [comment]
I think we should use native values if any of the values in a category don't match up with what we have in a mapping. So as an example, `[1, 2, 3]` is fine, but `[5, 6]` will just show as 5 and 6.
